libs/server/Storage/Session/Common/MemoryUtils.cs

```diff
@@ -38,6 +38,12 @@ public static class MemoryUtils
         /// <summary>.Net object avg. overhead for holding a hash set entry</summary>
         public const int HashSetEntryOverhead = 40;
 
+        /// <summary>.Net object overhead for priority queue</summary>
+        public const int PriorityQueueOverhead = 80;
```

Comment:
How do we calculate this? Is it possible to calculate this automatically from object definition?

Reply:
> How do we calculate this? Is it possible to calculate this automatically from object definition?

I don't know if it's right or not, I got these values from my simple testing from memory profiler. I don't know how but Github copilot also gave the same value. I am not confident with this value. memory profiler has varients when adding more than 1 elements. That why https://github.com/microsoft/garnet/pull/864#discussion_r1876632383 convercation thread is going on. CC: @badrishc @yrajas

Reply:
> How do we calculate this? Is it possible to calculate this automatically from object definition?

I don't know if it's right or not, I got these values from my simple testing from memory profiler. I don't know how but Github copilot also gave the same value. I am not confident with this value. memory profiler has varients when adding more than 1 elements. That why https://github.com/microsoft/garnet/pull/864#discussion_r1876632383 convercation thread is going on. CC: @badrishc @yrajas

Reply:
> How do we calculate this? Is it possible to calculate this automatically from object definition?

I don't know if it's right or not, I got these values from my simple testing from memory profiler. I don't know how but Github copilot also gave the same value. I am not confident with this value. memory profiler has varients when adding more than 1 elements. That why https://github.com/microsoft/garnet/pull/864#discussion_r1876632383 convercation thread is going on. CC: @badrishc @yrajas